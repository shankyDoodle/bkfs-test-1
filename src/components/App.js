import React from 'react'
import PropTypes from 'prop-types'

import {screenNames} from "../constants/appConstants";
import Home from './Home'

import Classification from './ClassificationView'
import Extraction from './Extraction'
import NavBar from './NavBar'
import {bindActionCreators} from "redux";
import * as myActions from "../actions";
import {connect} from "react-redux";
import {withStyles} from "@material-ui/core/styles";

class App extends React.Component {
    constructor(props) {
        super(props);
    }

    getScreenToRender = () => {
        let sScreenName = this.props.currentScreen;
        switch (sScreenName) {
            case screenNames.CLASSIFICATION:
                return <Classification/>
            case screenNames.EXTRACTION:
                return <Extraction/>
            default:
                return <Home/>
        }
    }

    render() {
        return (
            <div className={"appContainer"}>
                <NavBar/>
                <div className={"screenContainer"}>
                    {this.getScreenToRender()}
                </div>
            </div>
        )
    }
}

function mapStateToProps(state) {
    return state;
}

function mapDispatchToProps(dispatch) {
    // let appActions = bindActionCreators({ homeButtonClicked: myActions.handleScreenChanged });
    // return { ...appActions, dispatch };
    return {}
}

const ConnectedView = connect(mapStateToProps, null)(App);
export default ConnectedView

