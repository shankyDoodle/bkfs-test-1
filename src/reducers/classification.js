import {customerList, customerData} from "../mockdata/customerData";

export default {
    fetchCustomerList: function() {
        //implement server call here fro customer list
        return customerList;
    }
}