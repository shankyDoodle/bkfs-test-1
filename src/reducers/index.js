import getInitialState from "./initial-state";

import {
  HOME_BUTTON_CLICKED
} from '../actions'

function handleHomeButtonClicked(state, id) {
  console.log(state)
  return getInitialState()
}

export default (state = getInitialState(), action) => {
  switch (action.type) {
    case HOME_BUTTON_CLICKED:
      return handleHomeButtonClicked(state, action.id);

    default:
      return state
  }
}
