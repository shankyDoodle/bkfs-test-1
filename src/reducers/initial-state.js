import {screenNames} from "../constants/appConstants";

let initialState = {
    currentScreen: screenNames.HOME,
    customerList: {},
    selectedCustomers: [],
    documentTypes: {},
    selectedDocuments:[],
    customerData: {},
    classificationTableData: {},
    csvData:[],
    isScreenDirty: false,
    groupedDocumentElements:[],
    extractedSampleFile:null,
    textData:"",
    selectedDocumentSamples:{}
}

export const defaultState = initialState;

export const getInitialState = function () {
    return {
        ...initialState
    }
}