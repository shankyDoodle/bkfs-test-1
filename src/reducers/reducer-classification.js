import _ from 'lodash'

import {
    customerList,
    customerData,
    documentTypes
} from "../mockdata/mockData";

import * as appConstants from '../constants/appConstants';
import {Utils} from '../utils/utils';

export default {
    fetchCustomerData: function () {
        //implement server call here fro customer data
        return customerData;
    },

    setClassificationScreenOnLoadData: function(state, customerList, documentTypes){
        return {
            ...state,
            customerList,
            documentTypes
        }
    },

    handleClassificationDropDownOnBlur: function (state, dropdownType, selectedItems) {
        let oRet = {...state}
        if (dropdownType === appConstants.dropdownTypes.CUSTOMER_NAMES) {
            Object.assign(oRet, {selectedCustomers: selectedItems});
        } else {
            Object.assign(oRet, {selectedDocuments: selectedItems});
        }
        return oRet;
    },

    createCSVData: function(oTableData){
        let aData = [];

        let aHeaderData = oTableData.headerData;
        let aHeaderDataRow = [];
        for (let header of aHeaderData) {
            aHeaderDataRow.push(header.label);
        }
        aData.push(aHeaderDataRow);


        let aBodyData = oTableData.bodyData;
        _.forEach(aBodyData, function (oData) {
            let oRowData = oData.rowData;
            let row = [];
            for (let header of aHeaderData) {
                row.push(oRowData[header.label]);
            }
            aData.push(row);
        })

        return aData;
    },

    fetchSampleFiles: function(){
      //TODO: ideally fetch files from server
      return documentTypes;
    },

    handleClassificationCreateButtonCLicked: function (state, customerData, documentSamples) {
        let selectedDocuments = state.selectedDocuments;
        let selectedCustomers = state.selectedCustomers;

        let oCustomerList = state.customerList;
        let oDocumentTypes = state.documentTypes;

        let aHeaderData = [];
        aHeaderData.push({label: "Document Types"});
        for (let customer of selectedCustomers) {
            let label = oCustomerList[customer].label;
            aHeaderData.push({
                id: customer,
                label:label
            });
        }

        let aBodyData = [];
        for (let doc of selectedDocuments) {
            let sDocLabel = oDocumentTypes[doc].label;
            let values = {}
            values["Document Types"] = sDocLabel;

            for (let customer of selectedCustomers) {
                values[customer] = customerData[customer][sDocLabel]
            }
            let temp = {};
            temp["documentId"] = doc;
            temp["documentName"] = sDocLabel;
            temp["rowData"] = values
            temp["file"] = documentSamples[doc].data; //buffer data;
            temp["unusedFileData"] = documentSamples[doc];//backup
            aBodyData.push(temp);
        }


        let oTableData = {
            headerData: aHeaderData,
            bodyData: aBodyData
        }

        let aCSVData = this.createCSVData(oTableData);

        return {
            ...state,
            customerData,
            classificationTableData: oTableData,
            csvData: aCSVData
        }
    },


    handleTableCellDataChanged: function (state, customerId, docId, newVal) {

        //update local database
        let customerDataCloned = Utils.getDirtyData(state.customerData);
        customerDataCloned[customerId][docId] = newVal;

        //update table data
        let tableDataCloned = Utils.getDirtyData(state.classificationTableData);
        let bodyData = tableDataCloned.bodyData
        let editedRowIndex = _.findIndex(bodyData, function (o) {
            return o.documentId === docId;
        });
        let editedRowData = bodyData[editedRowIndex].rowData;
        editedRowData[customerId] = newVal;

        let aCSVData = this.createCSVData(tableDataCloned);

        return {
            ...state,
            isScreenDirty:true,
            csvData: aCSVData
        }
    },

    handleTableSaveDiscardClicked: function (state, buttonType) {
        let oRet = {...state}

        Object.assign(oRet, {isScreenDirty:false})
        if(buttonType === "save"){
            let customerData = state.customerData.clonedObject;
            let classificationTableData = state.classificationTableData.clonedObject;

            delete state.customerData.clonedObject
            delete state.customerData.isDirty
            delete state.classificationTableData.clonedObject;
            delete state.classificationTableData.isDirty;
            Object.assign(oRet, {customerData, classificationTableData});

            //TODO: handle any server calls to update Database
        }else {
            let customerData = state.customerData;
            let classificationTableData = state.classificationTableData;
            Object.assign(oRet, {customerData, classificationTableData});
        }

        delete state.customerData.clonedObject
        delete state.customerData.isDirty
        delete state.classificationTableData.clonedObject;
        delete state.classificationTableData.isDirty;

        return oRet;
    }

}