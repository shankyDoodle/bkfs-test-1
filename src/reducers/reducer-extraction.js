import _ from 'lodash'

import {
    documentTypes,
    groupedDocElements
} from "../mockdata/mockData";

import * as appConstants from '../constants/appConstants';
import {Utils} from '../utils/utils';

export default {
    fetchSampleFile: function (docName) {
        //TODO: ideally fetch files from server
        return "./pdfs/"+ documentTypes[docName];
    },

    fetchGroupedDocumentElementsByDocumentName: function (docName) {
        //TODO: ideally fetch files from server
        return groupedDocElements
    },

    setExtractionScreenOnLoadData: function (state, documentTypes) {
        return {
            ...state,
            documentTypes
        }
    },

    handleExtractionDropDownOnBlur: function (state, dropdownType, selectedItems) {
        return {
            ...state,
            selectedDocuments: [selectedItems]
        };
    },

    createSingleTextData: function (aGroupedDocumentData) {
        let sData = "";
        for (let oData of aGroupedDocumentData) {
            for (let dataElement of oData.dataElements) {
                sData+=dataElement+"\n";
            }
        }
        return sData;
    },

    handleExtractionCreateButtonCLicked: function (state) {
        let sSelectedDoc = state.selectedDocuments[0];
        let aGroupedDocumentData = this.fetchGroupedDocumentElementsByDocumentName(sSelectedDoc);
        let oFile = this.fetchSampleFile(sSelectedDoc);

        let textData = this.createSingleTextData(aGroupedDocumentData);

        return {
            ...state,
            textData,
            groupedDocumentElements: aGroupedDocumentData,
            extractedSampleFile: oFile
        }

    },

    moveWithinSameList: (list, startIndex, endIndex) => {
        const result = Array.from(list);
        const [removed] = result.splice(startIndex, 1);
        result.splice(endIndex, 0, removed);

        return result;
    },

    handleExtractionListDragEnd: function (state, source, destination) {
        let aGroups = Array.from(state.groupedDocumentElements);

        let sSourceGroupId = source.groupId
        let oFoundSourceGroup = aGroups.find(oG => oG.groupId === sSourceGroupId)
        let sDestGroupId = destination.groupId
        if (sSourceGroupId === sDestGroupId) {
            oFoundSourceGroup.dataElements = this.moveWithinSameList(oFoundSourceGroup.dataElements, source.index, destination.index);
        }else{
            let oFoundDestGroup = aGroups.find(oG => oG.groupId === sDestGroupId)
            let [removed] = oFoundSourceGroup.dataElements.splice(source.index, 1);
            oFoundDestGroup.dataElements.splice(destination.index, 0, removed);
        }

        let textData = this.createSingleTextData(aGroups);
        return {
            ...state,
            textData,
            groupedDocumentElements: aGroups
        }
    }
}