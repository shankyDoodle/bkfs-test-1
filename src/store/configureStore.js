import { createStore, applyMiddleware, compose } from 'redux'
import thunk from 'redux-thunk'
import { createLogger } from 'redux-logger'
// import api from '../middleware/api'
import rootReducer from '../reducers/reducer-index'
// import DevTools from '../containers/DevTools'

const configureStore = preloadedState => {
    const store = createStore(
        rootReducer,
        preloadedState,
        compose(
            applyMiddleware(thunk, createLogger())
        )
    )

    return store
}

export default configureStore
