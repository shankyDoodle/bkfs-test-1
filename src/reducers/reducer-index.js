import {getInitialState, defaultState} from "./initial-state";

import {screenNames} from "../constants/appConstants";
import * as appActions from '../actions'

import classification from "./reducer-classification";
import extraction from "./reducer-extraction";
import {HANDLE_EXTRACTION_DROP_DOWN_ON_BLUR, SET_EXTRACTION_SCREEN_ON_LOAD_DATA} from "../actions";



function resetToInitialState(oRet) {
    Object.assign(oRet, {...defaultState});
}
function handleScreenChanged(state, sScreenName) {
    let oRet = {...state}

    switch(sScreenName){
        case screenNames.HOME:
        default:
            resetToInitialState(oRet)
    }
    Object.assign(oRet, {currentScreen: sScreenName});
    return oRet
}

export default (state = getInitialState(), action) => {
    switch (action.type) {
        case appActions.HANDLE_SCREEN_CHANGED:
            return handleScreenChanged(state, action.screenName);

        case appActions.HANDLE_CLASSIFICATION_DROP_DOWN_ON_BLUR:
            return classification.handleClassificationDropDownOnBlur(state, action.dropdownButtonType, action.selectedItems);

        case appActions.HANDLE_CLASSIFICATION_CREATE_BUTTON_CLICKED:
            return classification.handleClassificationCreateButtonCLicked(state, action.customerData, action.documentSamples);

        case appActions.HANDLE_TABLE_CELL_DATA_CHANGED:
            return classification.handleTableCellDataChanged(state, action.customerName, action.docName, action.newVal);

        case appActions.HANDLE_TABLE_SAVE_DISCARD_CLICKED:
            return classification.handleTableSaveDiscardClicked(state, action.buttonType);

        case appActions.HANDLE_EXTRACTION_DROP_DOWN_ON_BLUR:
            return extraction.handleExtractionDropDownOnBlur(state, action.dropdownButtonType, action.selectedItems);

        case appActions.HANDLE_EXTRACTION_CREATE_BUTTON_CLICKED:
            return extraction.handleExtractionCreateButtonCLicked(state, action.groupedElements);

        case appActions.HANDLE_EXTRACTION_LIST_DRAG_END:
            return extraction.handleExtractionListDragEnd(state, action.source, action.destination);

        case appActions.SET_CLASSIFICATION_SCREEN_ON_LOAD_DATA:
            return classification.setClassificationScreenOnLoadData(state, action.customerList, action.documentTypes);

        case appActions.SET_EXTRACTION_SCREEN_ON_LOAD_DATA:
            return extraction.setExtractionScreenOnLoadData(state, action.documentTypes);

        default:
            return state
    }
}
