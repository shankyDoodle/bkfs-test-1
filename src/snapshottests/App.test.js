import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import App from '../components/App';
import {Provider} from "react-redux";
import {getInitialState} from "../reducers/initial-state";
import configureStore from "../store/configureStore";

const state = getInitialState();
const store = configureStore(state);

const component = renderer.create(
    <Provider store={store}><App /></Provider>,
);
test('renders learn react link', () => {

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
