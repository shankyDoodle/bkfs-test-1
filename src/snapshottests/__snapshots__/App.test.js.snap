// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="appContainer"
>
  <div
    className="navBar"
    style={
      Object {
        "backgroundColor": 800,
      }
    }
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          className="MuiTypography-root Connect(NavBar)-title-3 MuiTypography-h6"
        >
          BKFS
        </h6>
        <button
          aria-haspopup="true"
          aria-label="home"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z"
              />
            </svg>
          </span>
        </button>
      </div>
    </header>
  </div>
  <div
    className="screenContainer"
  >
    <div
      className="buttonWrapper"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained Connect(Home)-margin-99 MuiButton-containedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Classification
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained Connect(Home)-margin-99 MuiButton-containedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Extraction
        </span>
      </button>
    </div>
  </div>
</div>
`;
